webapps/frontend/app/viewmodels/site/vtemplates.js

```diff
@@ -110,14 +118,122 @@ define([
+                            deferred.resolve({
+                                data: data,
+                                loader: function(guid) {
+                                    return new VDisk(guid);
+                                },
+                                dependencyLoader: function(item) {
+                                    item.fetchTemplateChildrenGuids();
+                                    var vm, vMachineGuid = item.vMachineGuid();
+                                    if (vMachineGuid && (item.vMachine() === undefined || item.vMachine().guid() !== vMachineGuid)) {
+                                        if (!self.vMachineCache.hasOwnProperty(vMachineGuid)) {
+                                            vm = new VMachine(vMachineGuid);
+                                            self.vMachineCache[vMachineGuid] = vm;
+                                        }
+                                        item.vMachine(self.vMachineCache[vMachineGuid]);
+                                    }
+                                }
+                            });
+                        })
+                        .fail(function() { deferred.reject(); });
+                } else {
+                    deferred.resolve();
+                }
+            }).promise();
+        };
+        self.deleteVDT = function(guid) {
+            $.each(self.vDiskTemplates(), function(index, vd) {
+                if (vd.guid() === guid && vd.templateChildrenGuids().length === 0) {
+                    app.showMessage(
+                            $.t('ovs:vdisks.delete.warning', { what: vd.name() }),
+                            $.t('ovs:generic.areyousure'),
+                            [$.t('ovs:generic.no'), $.t('ovs:generic.yes')]
+                        )
+                        .done(function(answer) {
+                            if (answer === $.t('ovs:generic.yes')) {
+                                generic.alertInfo(
+                                    $.t('ovs:vdisks.delete.marked'),
+                                    $.t('ovs:vdisks.delete.markedmsg', { what: vd.name() })
+                                );
+                                api.post('vdisks/' + vd.guid() + '/delete_vtemplate')
+                                    .then(self.shared.tasks.wait)
+                                    .done(function() {
+                                        generic.alertSuccess(
+                                            $.t('ovs:vdisks.delete.done'),
+                                            $.t('ovs:vdisks.delete.donemsg', { what: vd.name() })
+                                        );
+                                    })
+                                    .fail(function(error) {
+                                        generic.alertError(
+                                            $.t('ovs:generic.error'),
+                                            $.t('ovs:generic.messages.errorwhile', {
```

Comment:
This codepath will not function as desired
Please use something similar to this:
```
$.t('ovs:generic.messages.errorwhile', {what: $.t('ovs:vdisks.set_as_template.error_msg', {what: vd.name(), error: error})}));

```